ElectronNET.API/App.cs

```diff
@@ -1527,6 +1527,34 @@ public void DockSetIcon(string image)
             BridgeConnector.Socket.Emit("appDockSetIcon", image);
         }
 
+        /// <summary>
+        /// A String which is the user agent string Electron will use as a global fallback.
+        /// </summary>
+        public string UserAgentFallback
+        {
+            get
+            {
+                return Task.Run<string>(() =>
+                {
+                    var taskCompletionSource = new TaskCompletionSource<string>();
+
+                    BridgeConnector.Socket.On("appGetUserAgentFallbackCompleted", (result) =>
+                    {
+                        BridgeConnector.Socket.Off("appGetUserAgentFallbackCompleted");
+                        taskCompletionSource.SetResult((string)result);
+                    });
+
+                    BridgeConnector.Socket.Emit("appGetUserAgentFallback");
+
+                    return taskCompletionSource.Task;
+                }).Result;
```

Comment:
Hmm .Result can causes [deadlocks](https://blog.stephencleary.com/2014/12/a-tour-of-task-part-6-results.html). The cleanest implementation would be:
public Task<string> GetUserAgentFallback 
public void SetUserAgentFallback

Or what do you guys think?

Reply:
I think it would be a problem with functions that are called regularly during application runtime. I would leave it at Properties.

Reply:
I think some application design is important too... otherwise we would only have methods..